app/main.py

```diff
@@ -1 +1,33 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        return f"{{Name: {self.name}, " \
+               f"Health: {self.health}, " \
+               f"Hidden: {self.hidden}}}"
+
+    @classmethod
+    def checking_in_alive(cls) -> dict:
+        cls.alive = [animal for animal in cls.alive if animal.health > 0]
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(other: Animal) -> None:
+        if isinstance(other, Herbivore) and not other.hidden:
+            other.health -= 50
+            Animal.checking_in_alive()
```

Comment:
If you have 1 million animals - you will need to go through this loop every time, when one of it dies. Just imagine the load. 
What about just removing dead animal from list?

Reply:
I've made some changes. Hope there're correct.

Reply:
The changed code basically is doing the same. You are looping through all animals to delete one of them. 
Why don't you just remove the exact `other` animal from alive list if it's health <= 0?

Reply:
Sorry. Now I've deleted method "checking_in_alive" and expanded functionality of the method "bite". It checks if the animal is dead at once.